Mage.Tests/src/test/java/org/mage/test/serverside/base/impl/CardTestPlayerAPIImpl.java

```diff
@@ -703,6 +704,14 @@ public void addPlane(Player player, Planes plane) {
         assertTrue("Can't put plane to game: " + plane.getClassName(), SystemUtil.putPlaneToGame(currentGame, player, plane.getClassName()));
     }
 
+    public void addEmblem(Player player, Emblem emblem) {
+        currentGame.addEmblem(
+                emblem,
+                emblem, // So this is a little ugly, but addEmblem is expecting a non-null source.
```

Comment:
So `GameImpl::addEmblem` does expect a non-null source, but it's only used for image generation which I don't think is applicable to TestPlayer.

Within `Emblem extends CommandObjectImpl`:
* getSourceObject() has three usages, two only apply on instanceof Commander, the third null checks
* getSourceId() nullchecks before returning.
* No other usages of sourceObject that I see.

So maybe it would make sense to call `currentGame.getState().addCommandObject(emblem)` directly? Only assigning the emblem characteristics that are relevant in the test environment?